scheduler/src/cook/compute_cluster.clj

```diff
@@ -156,3 +159,306 @@
                                 (map (fn [{:keys [config]}] (:compute-cluster-name config)))
                                 first)]
     (compute-cluster-name->ComputeCluster first-cluster-name)))
+
+(defn diff-map-keys
+  "Return triple of keys from two maps: [only in left, only in right, in both]"
+  [left right]
+  (data/diff (set (keys left)) (set (keys right))))
+
+(defn union-map-keys
+  "Return the union of the keys from two maps"
+  [left right]
+  (set/union (keys left) (keys right)))
+
+(defn compute-cluster-config-ent->compute-cluster-config
+  "Convert Datomic dynamic cluster configuration entity to an object"
+  [{:keys [compute-cluster-config/name
+           compute-cluster-config/template
+           compute-cluster-config/base-path
+           compute-cluster-config/ca-cert
+           compute-cluster-config/state
+           compute-cluster-config/state-locked?]}]
+  {:name name
+   :template template
+   :base-path base-path
+   :ca-cert ca-cert
+   :state (case state
+            :compute-cluster-config.state/running :running
+            :compute-cluster-config.state/draining :draining
+            :compute-cluster-config.state/deleted :deleted
+            state)
+   :state-locked? state-locked?})
+
+(defn compute-cluster-config->compute-cluster-config-ent
+  "Convert dynamic cluster configuration to a Datomic entity"
+  [{:keys [name template base-path ca-cert state state-locked?]}]
+  {:compute-cluster-config/name name
+   :compute-cluster-config/template template
+   :compute-cluster-config/base-path base-path
+   :compute-cluster-config/ca-cert ca-cert
+   :compute-cluster-config/state (case state
+                                   :running :compute-cluster-config.state/running
+                                   :draining :compute-cluster-config.state/draining
+                                   :deleted :compute-cluster-config.state/deleted
+                                   :state)
+   :compute-cluster-config/state-locked? state-locked?})
+
+(defn db-config-ents
+  ;TODO is it ok to fail to connect to the db and return empty list?
+  ;TODO maybe add "archived" flag and don't return archived configs for brevity
+  "Get the current dynamic cluster configurations from the database"
+  [db]
+  (let [configs (map #(d/entity db %)
+                     (d/q '[:find [?compute-cluster-config ...]
+                            :where
+                            [?compute-cluster-config :compute-cluster-config/name ?name]]
+                          db))]
+    (map-from-vals :compute-cluster-config/name configs)))
+
+;TODO: when all clusters have state, change the ComputeCluster protocol to add state operations
+(defn compute-cluster->compute-cluster-config
+  "Calculate dynamic cluster configuration from a compute cluster"
+  [{:keys [state-atom state-locked?-atom name]
```

Comment:
My vote is that it's fine to leave this as-is so that we don't spend significant time modifying the Mesos Compute Cluster, since it will be going away soon.